### ISSUE TITLE
New analyzer or extend exisiting one: Replace '.Count() > 1' with 'Skip(1).Any()'

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3530]
In the same spirit as the analyzer that suggests to use `.Any()` instead of `Count > 0`/`Count == 0` we could suggest `.Skip(1).Any()` in replacement of `Count > 1`/`Count == 1`. 

I am even thinking we could go a bit further and define a threshold (I don't think it's worth having it configurable) for which we can suggest the following change `Count() > X`/`Count() == X` -> `.Skip(X).Any()`.

WDYT?

**Comment by maxkoshevoi** [comment]
Do we need a threshold here? `Count()` always iterates over all collection, so it will always be more efficient to use `.Skip(X).Any()`

**Comment by Evangelink** [comment]
You are right! We should not report when this is not the LHS or RHS of a binary operation though:

```csharp
MyMethod(enum.Count()); // do not report
var x = enum.Count(); // do not report
if ((x = enum.Count()) > 10) // do not report
```

**Comment by Evangelink** [comment]
Also this rule should not report if the rule suggestion to use the `Count` property has raised an issue on the operation.

**Comment by mavasani** [comment]
So overall the suggestion looks good, but my primary concern is that the suggestion trades readability for performance, and might really not be critical enough for everyone. @Evangelink my suggestion would to be implement this with `RuleLevel.IdeHidden_BulkConfigurable` setting. We can still keep this enabled by default for FxCop analyzers package which is lot more aggressive.

**Comment by mavasani** [comment]
> Also this rule should not report if the rule suggestion to use the `Count` property has raised an issue on the operation.

Sounds good - I think it is best to implement this new rule within that existing analyzer.

**Comment by louis-z** [comment]
@Evangelink, @mavasani,

I'd like to look into this, if that's ok. Any pointers would be appreciated.

**Comment by mavasani** [comment]
Tagging @Jozkee and @jeffhandley. I think this issue should be moved to dotnet/runtime repo for triage/approval. Can one of you please transfer the issue?
